lib/plugins/aws/package/compile/events/alb/index.test.js

```diff
@@ -79,4 +81,267 @@ describe('AwsCompileAlbEvents', () => {
+          );
+        }));
+
+    it('supports setting value to "allow"', () =>
+      fixtures
+        .extend(
+          'functionDestinations',
+          serverlessConfiguration({ onUnauthenticatedRequest: 'allow' })
+        )
+        .then(fixturePath =>
+          runServerless({
+            cwd: fixturePath,
+            cliArgs: ['package'],
+          })
+        )
+        .then(serverless => {
+          ({ Resources: cfResources } = serverless.service.provider.compiledCloudFormationTemplate);
+          naming = serverless.getProvider('aws').naming;
+        })
+        .then(() => {
+          const albListenerRuleLogicalId = naming.getAlbListenerRuleLogicalId('trigger', 1);
+          const rule = cfResources[albListenerRuleLogicalId];
+
+          expect(rule.Properties.Actions[0].AuthenticateCognitoConfig).to.deep.equal(
+            baseAuthenticateCognitoConfig({ OnUnauthenticatedRequest: 'allow' })
+          );
+        }));
+
+    it('supports setting value to "deny"', () =>
+      fixtures
+        .extend(
+          'functionDestinations',
+          serverlessConfiguration({ onUnauthenticatedRequest: 'deny' })
+        )
+        .then(fixturePath =>
+          runServerless({
+            cwd: fixturePath,
+            cliArgs: ['package'],
+          })
+        )
+        .then(serverless => {
+          ({ Resources: cfResources } = serverless.service.provider.compiledCloudFormationTemplate);
+          naming = serverless.getProvider('aws').naming;
+        })
+        .then(() => {
+          const albListenerRuleLogicalId = naming.getAlbListenerRuleLogicalId('trigger', 1);
+          const rule = cfResources[albListenerRuleLogicalId];
+
+          expect(rule.Properties.Actions[0].AuthenticateCognitoConfig).to.deep.equal(
+            baseAuthenticateCognitoConfig({ OnUnauthenticatedRequest: 'deny' })
+          );
+        }));
+
+    it('supports setting value to "authenticate"', () =>
+      fixtures
+        .extend(
+          'functionDestinations',
+          serverlessConfiguration({ onUnauthenticatedRequest: 'authenticate' })
+        )
+        .then(fixturePath =>
+          runServerless({
+            cwd: fixturePath,
+            cliArgs: ['package'],
+          })
+        )
+        .then(serverless => {
+          ({ Resources: cfResources } = serverless.service.provider.compiledCloudFormationTemplate);
+          naming = serverless.getProvider('aws').naming;
+        })
+        .then(() => {
+          const albListenerRuleLogicalId = naming.getAlbListenerRuleLogicalId('trigger', 1);
+          const rule = cfResources[albListenerRuleLogicalId];
+
+          expect(rule.Properties.Actions[0].AuthenticateCognitoConfig).to.deep.equal(
+            baseAuthenticateCognitoConfig({ OnUnauthenticatedRequest: 'authenticate' })
+          );
+        }));
+
+    it('takes precedence over allowUnauthenticated set "deny"', () =>
```

Comment:
Let's remove that test. I think that testing that `onUnauthenticatedRequest` takes precedence over `allowUnauthenticated` shouldn't require more than one test (and anyway such tests will be removed with removal of support for `allowUnauthenticated`)